tools/run_tests/python_utils/start_port_server.py

```diff
@@ -73,12 +73,15 @@ def start_port_server():
             # Working directory of port server needs to be outside of Jenkins
             # workspace to prevent file lock issues.
             tempdir = tempfile.mkdtemp()
-            port_server = subprocess.Popen(
-                args,
-                env=env,
-                cwd=tempdir,
-                creationflags=0x00000008,  # detached process
-                close_fds=True)
+            if sys.version_info.major == 2:
+                creationflags = 0x00000008  # detached process
+            else:
+                creationflags = 0  # DETACHED_PROCESS doesn't seem to work with python3
```

Comment:
This is surprising. The docs indicate that DETACHED_PROCESS wasn't actually even introduced until 3.7. Have you tried using [the symbol](https://docs.python.org/3/library/subprocess.html#subprocess.DETACHED_PROCESS)?

Reply:
Yes, I tried that and it didn't work either (not sure why, but it tried several times and whenever I used it, the port server process just failed to start).
I think the detached_process options was more important in the past (when we were using jenkins and long-lived worker VMs), because we needed port server to survive the CI build - this is not the case anymore.
I tried the port server locally and it worked, so I'm not too afraid to leave as is.